ModList/UI/ModListFlowCoordinator.cs

```diff
@@ -1,87 +1,120 @@
 
-                listController.DidSelectPlugin += HandleSelectPlugin;
-                controlsController.OnListNeedsRefresh += HandleListNeedsRefresh;
-                controlsController.OnChangeNeedsConfirmation += modalsController.QueueChange;
-
-                PushViewControllerToNavigationController(naviController, modalsController, immediately: true);
+                modListViewController.DidSelectPlugin += HandleSelectPlugin;
+                modControlsViewController.OnListNeedsRefresh += HandleListNeedsRefresh;
+                modControlsViewController.OnChangeNeedsConfirmation += modalPopupViewController.QueueChange;
             }
-            catch (Exception e)
+            catch (Exception ex)
             {
-                Logger.log.Error("Error activating TestFlowCoordinator");
-                Logger.log.Error(e);
+                siraLog.Error(ex);
             }
         }
 
-        protected override void DidDeactivate(DeactivationType deactivationType)
+        protected override void DidDeactivate(bool removedFromHierarchy, bool screenSystemDisabling)
         {
-            controlsController.OnChangeNeedsConfirmation -= modalsController.QueueChange;
-            controlsController.OnListNeedsRefresh -= HandleListNeedsRefresh;
-            listController.DidSelectPlugin -= HandleSelectPlugin;
-            base.DidDeactivate(deactivationType);
+            modControlsViewController.OnChangeNeedsConfirmation -= modalPopupViewController.QueueChange;
+            modControlsViewController.OnListNeedsRefresh -= HandleListNeedsRefresh;
+            modListViewController.DidSelectPlugin -= HandleSelectPlugin;
+
+            base.DidDeactivate(removedFromHierarchy, screenSystemDisabling);
         }
 
         private void HandleSelectPlugin(PluginInformation plugin)
         {
-            Logger.log.Info($"Mod list selected plugin {plugin.Plugin} ({plugin.State})");
-
-            infoController.SetPlugin(plugin);
-            controlsController.SetPlugin(plugin);
-            if (!infoController.Activated)
-                PushViewControllerToNavigationController(naviController, infoController);
+            siraLog.Info($"Mod list selected plugin {plugin.Plugin} ({plugin.State})");
+            modInfoViewController.SetPlugin(plugin);
+            modControlsViewController.SetPlugin(plugin);
         }
 
         private void HandleListNeedsRefresh()
         {
-            listController.ReloadViewList();
+            modListViewController.ReloadViewList();
         }
 
-        protected override void BackButtonWasPressed(ViewController topViewController)
+        protected override void BackButtonWasPressed(ViewController _)
         {
-            base.BackButtonWasPressed(topViewController);
-            ParentFlowCoordinator.DismissFlowCoordinator(this);
+            // Check whether there's a transaction going on and commit :eyes:
+            // TODO: Guess it's a good idea to add a confirmation here as well...
```

Comment:
It would probably also be worth adding an additional confirmation if they're disabling ModList.

Reply:
Can I just add it to the TODO or did I have to actually implement this as well before the PR can be merged?

Reply:
As a TODO is probably fine